### PR TITLE
Hookify + Type remote windows

### PR DESCRIPTION
**Pull request by chrisnojima**
This PR removes all the remotewindow based HOCs and replaces them with hooks. 
Additionally this allows the entire flow to be correctly typed

- [x] remove `SyncAvatarProps`, `SyncBrowserWindow` , `SyncProps`
- [x] add `useBrowserWindow`, `useSerializeProps`
- [x] smarter deduping to reduce rpc traffic
- [x] remove `Container.remoteConnect`
- [x] add `Container.useRemoteStore`
- [x] remove `remote/util/autoResize`
- [x] remove `common-adapters/usernames/remote-container`
- [x] fix 2 incorrect dispatch firing on creation in `menubar/index.desktop`
- [x] remote `bio/remote-container` and `assertions/remote-container`
- [x] added `util/map/mapFilterByKey`

Some small changes to tracker as in master `inTracker` is actually always `false` in stories

### CODE REVIEW

File: shared/menubar/remote-container.desktop.tsx [hunk 1]
```diff
@@ -1,89 +1,93 @@
...
-    dispatch(FsGen.createLoadPathMetadata({path: Types.stringToPath('/keybase/private' + name)})),
-  _showUser: (username: string) => {
-    const link = urlHelper('user', {username})
-    link && openUrl(link)
-  },
-  logIn: () => {
-    dispatch(ConfigGen.createShowMain())
-    dispatch(RouteTreeGen.createNavigateAppend({path: [loginTab]}))
-  },
-  onHideDiskSpaceBanner: dispatch(FsGen.createShowHideDiskSpaceBanner({show: false})),
```
Comment (chrisnojima) [inline shared/menubar/remote-container.desktop.tsx]: actual bug!

File: shared/menubar/remote-container.desktop.tsx [hunk 2]
```diff
@@ -1,89 +1,93 @@
...
-  showInFinder: () => dispatch(FsGen.createOpenPathInSystemFileManager({path: FsConstants.defaultPath})),
-  updateNow: isWindows || isDarwin ? () => dispatch(ConfigGen.createUpdateNow()) : undefined,
-  waitForKbfsDaemon: throttle(() => dispatch(FsGen.createWaitForKbfsDaemon()), 1000 * 5),
-})
-
-const mergeProps = (stateProps, dispatchProps, ownProps) => {
-  return {
-    ...stateProps,
-    ...dispatchProps,
-    onRetrySync: dispatchProps._onRetrySync(stateProps.username),
```
Comment (chrisnojima) [inline shared/menubar/remote-container.desktop.tsx]: actual bug!

File: shared/desktop/remote/util.desktop.tsx [hunk 3]
```diff
@@ -1,60 +1,6 @@
 import * as SafeElectron from '../../util/safe-electron.desktop'
 import {TypedActions} from '../../actions/typed-actions-gen'
 
-export const autoResize = () => {
```
Comment (chrisnojima) [inline shared/desktop/remote/util.desktop.tsx]: only used by pinentry and its flakey, just use a fixed size

File: shared/menubar/remote-container.desktop.tsx (hunk 1 above)
Comment (jakob223) [reply shared/menubar/remote-container.desktop.tsx]: Ha, I guess this banner was just never showing >.<

File: shared/menubar/remote-container.desktop.tsx (hunk 2 above)
Comment (jakob223) [reply shared/menubar/remote-container.desktop.tsx]: I don't think this was an actual bug? you mean the missing `() =>`? It's not the way we usually do things but the definition in `dispatchProps` was `(name) => () => ...`

File: shared/tracker2/bio/container.tsx [hunk 4]
```diff
@@ -1,51 +1,48 @@
...
-    return {
-      ...common,
-      bio: d.bio,
-      followThem: Constants.followThem(state, ownProps.username),
-      followersCount: d.followersCount,
-      followingCount: d.followingCount,
-      followsYou: Constants.followsYou(state, ownProps.username),
-      fullname: d.fullname,
-      location: d.location,
+    if (d.state === 'notAUserYet') {
```
Comment (jakob223) [inline shared/tracker2/bio/container.tsx]: Is this a special value that is used when `getDetails` doesn't find anything? Can we make it a constant in `Constants`?

File: shared/desktop/remote/use-serialize-props.desktop.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+  const throttledSend = React.useRef(
+    throttle(
+      (p: ProxyProps, forceUpdate: boolean) => {
+        const lastToSend = forceUpdate ? {} : lastSent.current
+        const serialized = serializer(p)
+        const toSend = {...serialized}
+        // clear undefineds / exact dupes
+        Object.keys(toSend).forEach(k => {
+          if (toSend[k] === undefined || JSON.stringify(toSend[k]) === JSON.stringify(lastToSend?.[k])) {
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/desktop/remote/use-serialize-props.desktop.tsx]: this is really bulletproof and we want it per key at the top level so, it seems fine. its not slow on widget which has the biggest payload

File: shared/tracker2/bio/container.tsx (hunk 4 above)
Comment (chrisnojima) [reply shared/tracker2/bio/container.tsx]: this is typed, its `DetailsState` (and unchanged). TS will complain if its not part of the enum

**Pull request merged**